gpio.py

```diff
@@ -102,25 +105,38 @@ def cleanup(pin=None, assert_exists=False):
 
 
 @_verify
-def setup(pin, mode, pullup=None, initial=False):
+def setup(pin, mode, pullup=None, initial=False, active_low=None):
     '''Setup pin with mode IN or OUT.
 
     Args:
         pin (int):
         mode (str): use either gpio.OUT or gpio.IN
         pullup (None): rpio compatibility. If anything but None, raises
             value Error
-        pullup (bool, optional): Initial pin value. Default is False
+        initial (bool, optional): Initial pin value. Default is False
+        active_low (bool, optional): Set the pin to active low. Default
+            is None which leaves things as configured in sysfs
     '''
     if pullup is not None:
         raise ValueError("sysfs does not support pullups")
 
     if mode not in (IN, OUT, LOW, HIGH):
         raise ValueError(mode)
 
+    if not type(active_low) == bool:
+        raise ValueError("active_low argument must be True or False")
+
+    if active_low is not None:
+        log.debug("Set active_low {0}: {1}".format(pin, active_low))
+        f_active_low = _open[pin].active_low
+        if active_low:

```

Comment:
instead of if/else

`_write(f_active_low, int(active_low))`

You've already checked extensively that it is a boolean type!

Reply:
good point. changed.